src/main/java/org/carlmontrobotics/robotcode2023/commands/ArmPeriodic.java

```diff
@@ -29,47 +29,67 @@ public ArmPeriodic(Arm armSubsystem, DoubleSupplier arm, DoubleSupplier wrist) {
   @Override
-  public void initialize() {}
+  public void initialize() {
+  }
 
   // Called every time the scheduler runs while the command is scheduled.
   @Override
   public void execute() {
     double[] speeds = getRequestedSpeeds();
-    armSubsystem.driveArm(speeds[0], 0);
-    armSubsystem.driveWrist(speeds[1], 0);
-    armSubsystem.setArmTarget(armSubsystem.getArmPos());
-    armSubsystem.setWristTarget(armSubsystem.getWristPos());
+    // if driven by joysticks, run only feedforward
+    if (speeds[0] == 0) 
+      armSubsystem.driveArm(0, 0, true);
+    else {
+      armSubsystem.driveArm(speeds[ARM], 0, false);
+      armSubsystem.setArmTarget(armSubsystem.getArmPos());
+    }
+    if (speeds[1] == 0)
+      armSubsystem.driveWrist(0, 0, true);
+    else {
+      armSubsystem.driveWrist(speeds[WRIST], 0, false);
+      armSubsystem.setWristTarget(armSubsystem.getWristPos());
+    }
   }
 
-  public double[] getRequestedSpeeds() {double rawArm, rawWrist, deltaT;
+  // Copy and pasted from drivetrain, handles input from joysticks
+  public double[] getRequestedSpeeds() {
+    double rawArm, rawWrist, deltaT;
     deltaT = .05;
-    // Sets all values less than or equal to a very small value (determined by the idle joystick state) to zero.
-    // Used to make sure that the robot does not try to change its angle unless it is moving,
-    if (Math.abs(arm.getAsDouble()) <= Constants.OI.JOY_THRESH) rawArm = 0.0;
-    else rawArm = MAX_FF_VEL * arm.getAsDouble();
-    if (Math.abs(wrist.getAsDouble()) <= Constants.OI.JOY_THRESH) rawWrist = 0.0;
-    else rawWrist = MAX_FF_VEL * wrist.getAsDouble();
+    // Sets all values less than or equal to a very small value (determined by the
+    // idle joystick state) to zero.
+    // Used to make sure that the robot does not try to change its angle unless it
+    // is moving,
+    if (Math.abs(arm.getAsDouble()) <= Constants.OI.JOY_THRESH)
+      rawArm = 0.0;
+    else
+      rawArm = MAX_FF_VEL[ARM] * arm.getAsDouble();
+    if (Math.abs(wrist.getAsDouble()) <= Constants.OI.JOY_THRESH)
+      rawWrist = 0.0;
+    else
+      rawWrist = MAX_FF_VEL[WRIST] * wrist.getAsDouble();
 
-    double targetAccelerationArm = (rawArm - currentArm)/deltaT;
-    double targetAccelerationWrist = (rawWrist - currentWrist)/deltaT;
-    if (targetAccelerationArm >= MAX_FF_ACCEL) {
-      targetAccelerationArm *= MAX_FF_ACCEL/targetAccelerationArm;
+    double targetAccelerationArm = (rawArm - currentArm) / deltaT;
+    double targetAccelerationWrist = (rawWrist - currentWrist) / deltaT;
+    if (targetAccelerationArm >= MAX_FF_ACCEL[ARM]) {
+      targetAccelerationArm *= MAX_FF_ACCEL[ARM] / targetAccelerationArm;
     }
-    if (targetAccelerationWrist >= MAX_FF_ACCEL) {
-      targetAccelerationWrist *= MAX_FF_ACCEL/targetAccelerationWrist;
+    if (targetAccelerationWrist >= MAX_FF_ACCEL[WRIST]) {
+      targetAccelerationWrist *= MAX_FF_ACCEL[WRIST] / targetAccelerationWrist;
     }
```

Comment:
Unless I'm missing something, the only difference between these lines a couple calls to `Math.max()` are that these lines can introduce divide-by-zero errors.

Reply:
The current implementation also only limits the acceleration in one direction.